src/Razor/Microsoft.AspNetCore.Razor.Language/src/StringSegment.cs

```diff
@@ -160,15 +159,7 @@ public bool Equals(string text)
     /// <param name="comparisonType">One of the enumeration values that specifies the rules to use in the comparison.</param>
     /// <returns><code>true</code> if the specified <see cref="string"/> is equal to the current <see cref="StringSegment"/>; otherwise, <code>false</code>.</returns>
     public bool Equals(string text, StringComparison comparisonType)
-    {
-        var textLength = text.Length;
```

Comment:
The null ref is here. The `Equals(StringSegment)` is null-safe, so using that which also helps remove duplicated code.